django/db/models/fields/generated.py

```diff
@@ -30,6 +30,9 @@ def __init__(self, *, expression, output_field, db_persist=None, **kwargs):
         self.expression = expression
         self.output_field = output_field
         self.db_persist = db_persist
+
+        if kwargs.get("choices", None) is None and output_field.choices is not None:
+            kwargs["choices"] = output_field.choices
```

Comment:
I wonder if `choices` should be an acceptable argument for `GeneratedField`s :thinking: Can we check how forms and `clean()` behave in such configurations?

Reply:
`GeneratedField`s are ignored in `full_clean()`
I added to the test to show that the `GeneratedField` can be something that isn't one of the choices (as the choices go up to 5) and played around with it as a readonly field in the admin, which looks ok to me

So I think it's ok - is there a specific concern you want me to check for?

<details><summary>Admin screenshots</summary>

![image](https://github.com/user-attachments/assets/3393dfc4-9d58-44d0-9b93-fa37d477da94)

![Screenshot from 2024-09-01 16-11-57](https://github.com/user-attachments/assets/b7b7596a-19d6-4e3e-8174-c06cfcf833b8)

</details>